File: HW5/heap/testHeap.py
import heap as h

heap = h.Heap()
opNum = 0
heapNum = 0
numCount = 0
opCount = 0
key = []
element = []
index = 0
with open("input.txt") as infile:
	for line in infile:
		if opNum == 0:
			opNum = int(line)
		elif heapNum == 0:
			heapNum = int(line)
		elif numCount < heapNum:
			l = line.split()
			key.append( int( l[0] ) )
			element.append( l[1] )
			numCount += 1
		elif opCount < opNum:
			if int(line) == 1:
				heap.push(key[index], element[index])
				index += 1
			else:
				heap.pop()
			opCount += 1			
heap.printArray()
heap.printByPopping()

File: HW5/heap/testHeap.cpp
#include <iostream>
#include <fstream>

using namespace std;

#include "heap.h"

int main()
{

    Heap<int, char> heap;

    ifstream file("input.txt");

    int opNum;
    int heapNum;
    file >> opNum >> heapNum;
    int *key = new int[heapNum];
    char *element = new char[heapNum];

    for (int i=0; i<heapNum; ++i) {
        file >> key[i];
        file >> element[i];
    }
    int index = 0;
    for (int i=0; i<opNum; ++i) {
        int op;
        file >> op;
        if (op == 1) {
            heap.push(key[index], element[index]);
            ++index;
        }
        else 
            heap.pop();
    }

    file.close();
    delete []element;
    delete []key;

    heap.printArray();

    heap.printByPopping();

    return 0;
}


